docs/porting/standard_pin_names/pin_names_porting.md

```diff
@@ -0,0 +1,234 @@
+
+The tool will automatically run the appropriate test suites by reading the supported form factors from the target definition in `targets.json`. If this information is not available, the tool will only run the `generic` test suite by default.
+
+You can manually choose which test suites to run using the `-n` flag:
+`pinvalidate.py -t L475 -n generic`
+`pinvalidate.py -t L475 -n generic,arduino`
+
+The scripts supports several output formats. The default output format is `prettytext`. You can choose the output format with the `-o` flag:
+`pinvalidate.py -t L4S5I -o json`
+`pinvalidate.py -t L4S5I -o html`
+
+You may also want to use the `-w` flag to write the output to a file when choosing HTML or JSON output:
+`pinvalidate.py -t L4S5I -o html -w output.html`
+
+The `prettytext` format supports four levels of detail/verbosity, selected with the usual `-v` flag:
+`pinvalidate.py -t L4S5I` (level 0)
+`pinvalidate.py -t L4S5I -v` (level 1)
+`pinvalidate.py -t L4S5I -vvv` (level 3)
+* At the default level (0), the script will output a short summary table showing a PASS/FAIL outcome for each of the validated files/targets.
+* At level 1, the script will output a summary table showing a PASS/FAIL outcome for each test suite ran against a target.
+* At level 2, the script will output a table showing a PASS/FAIL outcome for each test case ran against a target.
+* At level 3, the script will output all of the above information, with details of each individual error.
+
+When HTML output is chosen, the script will output the report as an interactive HTML table that you can view in a web browser. Each row is the output from one test case, with a PASS/FAIL outcome. You can view the details of any errors by expanding the last cell of the row.
+
+JSON and HTML output always include the highest level of detail.
+
+To verify that the target markers in PinNames.h files are valid, run `pinvalidate.py -m`.
+
+### Using Greentea
+
+There is also a Greentea test to facilitate runtime compliance testing on actual hardware. This test will typically be run in the final stages of the migration, after the `PinNames.h` file is already passing the `pinvalidate.py` test. The Greentea test checks that every required pin is defined and then checks that the pins support the features specified in the Arduino standard by making some basic API calls. For example, for pins that are supposed to provide PWM functionality, the test case will configure them as such and then do some basic calls to generate a waveform on those pins.
+
+After setting up your environment to be able to run Greentea tests, you can compile run the standard pin names compliance test suites:
+
+`mbed test -t GCC_ARM -m DISCO_L475VG_IOT01A -n "*pin_names*" --compile`
+
+`mbed test -t GCC_ARM -m DISCO_L475VG_IOT01A -n *pin_names* --run`
+
+You can expect the following output:
+
+```(p3venv) geopsi01@C02WX27NHV2R mbed-os % mbed test -t GCC_ARM -m DISCO_L475VG_IOT01A -n "*pin_names*" --run
```

Comment:
```suggestion
``` mbed-os % mbed test -t GCC_ARM -m DISCO_L475VG_IOT01A -n *pin_names*
```

Reply:
Keep it simple: `mbed test ...`